src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs

```diff
@@ -129,6 +133,19 @@ public void Commit(ILogger log, bool forceWrite)
                 }
             }
 
+            // Always write the tool lock files
+            foreach (var toolRestoreResult in ToolRestoreResults)
+            {
+                if (toolRestoreResult.LockFilePath != null)
+                {
+                    log.LogDebug($"Writing tool lock file to disk. Path: {toolRestoreResult.LockFilePath}");
+
+                    var lockFileDirectory = Path.GetDirectoryName(toolRestoreResult.LockFilePath);
```

Comment:
The main lock file is only written out if it changes to avoid modifying the file timestamp. This allows incremental builds to skip rebuilding in some cases. I'm not sure if that is a scenario here or not. It could be done in another change also.